python_modules/dagster/dagster/_core/storage/db_io_manager.py

```diff
@@ -143,7 +143,8 @@ def handle_output(self, context: OutputContext, obj: object) -> None:
             self._db_client.ensure_schema_exists(context, table_slice, conn)
             self._db_client.delete_table_slice(context, table_slice, conn)
 
-            handler_metadata = self._handlers_by_type[obj_type].handle_output(
+            handler = next(filter(lambda x: issubclass(obj_type, x[0]), self._handlers_by_type.items()))[1]
```

Comment:
This could raise various uncaught exceptions, but I think the same applies on `self._handlers_by_type[obj_type]`.